### PR TITLE
CsvDataSet no longer globs files in init.

### PR DESCRIPTION
**Pull request by brandondutra**

### CODE REVIEW

**Comment by brandondutra** [comment]
ok, updated the class to not always glob the input file list

File: datalab/mlalpha/_dataset.py [hunk 1]
```diff
@@ -59,21 +60,34 @@ def __init__(self, files, schema=None, schema_file=None):
...
+
+  @property
+  def _input_files(self):
+    """Returns the file list that was given to this class without globing files."""
+    return self._input_files
 
   @property
   def files(self):
-    return self._files
+    if self._glob_files:
```
Comment (qimingj) [inline datalab/mlalpha/_dataset.py]: Is:

if not self._glob_files:
  self._glob_files += ...

return self._glob_files

better?

File: datalab/mlalpha/_dataset.py [hunk 2]
```diff
@@ -59,21 +60,34 @@ def __init__(self, files, schema=None, schema_file=None):
...
+      return self._glob_files
+
+    for file in self._input_files:
+      # glob_files() returns unicode strings which doesn't make DataFlow happy. So str().
+      self._files += [str(x) for x in ml.util._file.glob_files(file)]
```
Comment (qimingj) [inline datalab/mlalpha/_dataset.py]: Should be self._glob_files?

File: datalab/mlalpha/_dataset.py (hunk 1 above)
Comment (brandondutra) [inline datalab/mlalpha/_dataset.py]: ops, yes

**Comment by brandondutra** [comment]
ok, now done :)

**Pull request merged**